keystorage/hashicorp/src/testFixtures/java/tech/pegasys/signers/hashicorp/dsl/HashicorpLocalNode.java

```diff
@@ -0,0 +1,276 @@
+  private final AtomicInteger port = new AtomicInteger(0);
+
+  // initialized after start
+  private HashicorpVaultCommands hashicorpVaultCommands;
+  private Future<ProcessResult> serverFuture;
+  private String vaultToken;
+
+  public HashicorpLocalNode(
+      final Path vaultBinary, final SelfSignedCertificate serverTlsCertificate) {
+    this.vaultBinary = vaultBinary;
+    this.serverTlsCertificate = Optional.ofNullable(serverTlsCertificate);
+  }
+
+  public HashicorpLocalNode(final Path vaultBinary) {
+    this(vaultBinary, null);
+  }
+
+  void start()
+      throws IOException, CertificateEncodingException, TimeoutException, InterruptedException {
+    startVaultServer();
+    hashicorpVaultCommands = new HashicorpVaultCommands(vaultBinary.toString(), vaultUrl());
+
+    final HashicorpVaultTokens hashicorpVaultTokens = initVault();
+    unseal(hashicorpVaultTokens.getUnsealKey());
+    login(hashicorpVaultTokens.getRootToken());
+    enableHashicorpKeyValueV2Engine();
+  }
+
+  // Start vault server. Sets port variable once started.
+  private void startVaultServer() throws IOException, CertificateEncodingException {
+    LOG.info("Starting vault server");
+    final AtomicBoolean serverStarted = new AtomicBoolean(false);
+    serverFuture =
+        new ProcessExecutor(vaultBinary.toString(), "server", "-config=" + getVaultServerConfig())
+            .readOutput(true)
+            .redirectOutput(
+                new LogOutputStream() {
+                  @Override
+                  protected void processLine(final String line) {
+                    LOG.info(line);
+
+                    // process Vault port (i.e. cluster port - 1)
+                    if (line.contains("cluster address")) {
+                      final Matcher matcher = portPattern.matcher(line);
+                      matcher.find();
+                      port.set(Integer.parseInt(matcher.group(1)) - 1);
+                    }
+
+                    if (line.contains("Vault server started!")) {
+                      serverStarted.set(true);
+                    }
+                  }
+                })
+            .start()
+            .getFuture();
+
+    Awaitility.await().atMost(5, TimeUnit.SECONDS).untilTrue(serverStarted);
+  }
+
+  @Override
+  public void shutdown() {
+    if (serverFuture != null) {
+      LOG.info("Shutting down Vault server Process...");
+      serverFuture.cancel(true);
+      serverFuture = null;
+    } else {
+      LOG.info("Vault server process is not running.");
+    }
+  }
+
+  @Override
+  public String getVaultToken() {
+    return vaultToken;
+  }
+
+  @Override
+  public String getHost() {
+    return host;
+  }
+
+  @Override
+  public int getPort() {
+    return port.intValue();
+  }
+
+  @Override
+  public Optional<SelfSignedCertificate> getServerCertificate() {
+    return serverTlsCertificate;
+  }
+
+  @Override
+  public void addSecretsToVault(final Map<String, String> entries, final String path) {
```

Comment:
nit: shouldn't this be common to both execution models of Hashicorp?

Reply:
both of them are using same command set (via HashicorpVaultCommands), however, exact execution mechanism is different. The goal was to keep existing usage intact.